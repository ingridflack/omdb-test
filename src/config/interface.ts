export interface IMovie {
  imdbID: string;
  Poster: string;
  Title: string;
  Year: string;
  Runtime: string;
  Rated: string;
  imdbRating: string;
  Plot: string;
  Actors: string;
  Genre: string;
  Director: string;
}
