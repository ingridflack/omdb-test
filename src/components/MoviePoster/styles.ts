import styled from "styled-components";

export const Image = styled.img`
  height: 100%;
  object-fit: cover;
  border-radius: 8px;
  max-width: 100%;
`;
