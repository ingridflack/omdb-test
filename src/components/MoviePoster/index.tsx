import { IMovie } from "../../config/interface";
import { Image } from "./styles";

interface IMoviePoster {
  movie: IMovie;
}

const MoviePoster: React.FC<IMoviePoster> = ({ movie }) => {
  const source =
    movie.Poster === "N/A"
      ? "https://via.placeholder.com/320x460?text=Image+not+found"
      : movie.Poster;

  return <Image src={source} alt={movie.Title} />;
};

export default MoviePoster;
