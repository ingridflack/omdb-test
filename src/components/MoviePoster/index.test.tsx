import { shallow } from "enzyme";
import MoviePoster from ".";
import { IMovie } from "../../config/interface";

const MOVIE: IMovie = {
  imdbID: "ID Test",
  Poster: "Poster Test",
  Title: "Title Test",
  Year: "1996",
  Runtime: "100 min",
  Rated: "Rated Test",
  imdbRating: "Rating Test",
  Plot: "Plot Test",
  Actors: "Actors Test",
  Genre: "Genre Test",
  Director: "Director Test",
};

const MOVIE_WITHOUT_POSTER = {
  ...MOVIE,
  Poster: "N/A",
};

describe("MoviePoster", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<MoviePoster movie={MOVIE} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot with N/A poster image", () => {
    const wrapper = shallow(<MoviePoster movie={MOVIE_WITHOUT_POSTER} />);
    expect(wrapper).toMatchSnapshot();
  });
});
