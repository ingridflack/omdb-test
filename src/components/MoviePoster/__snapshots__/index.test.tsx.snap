// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePoster should match snapshot 1`] = `
<styled.img
  alt="Title Test"
  src="Poster Test"
/>
`;

exports[`MoviePoster should match snapshot with N/A poster image 1`] = `
<styled.img
  alt="Title Test"
  src="https://via.placeholder.com/320x460?text=Image+not+found"
/>
`;
